CONTRIBUTING.md

```diff
@@ -0,0 +1,276 @@
+1. The Django web app/http API process, providing:
+    - A web UI for researchers to design and manage experiments
+    - A web UI for participants to discover experiments
+    - An HTTP/JSON API consumed by built experiments
+2. An offline Celery process (connected by message queue) that builds and deploys experiments (web archives)
+3. An offline Celery process (connected by message queue) that sends emails and does some rudimentary cleanup
+
+The processes listed above also has three major *in-network* service dependencies:
+1. **Postgres** for relational data storage, used by all processes
+2. **RabbitMQ** as a message queue to communicate between process #1 and the two offline task runner processes
+3. **Docker** to provide a container environment for building experiments (used by process #2)
+
+And as if that wasn't enough, three major *out of network* service dependencies:
+1. **Amazon S3** for storing video
+2. **Google Cloud Storage** for storing experiment web archives and other static files
+3. [**Pipe**][pipe] to relay video streamed from experiments to Amazon S3
+
+For the kicker, if you want to debug the video renaming webhook, you'll need to install and run `ngrok`, and [add
+a new webhook target][pipe-add-webhook] in the Pipe web UI with the freshly generated URL. _Yikes_. Thankfully, you
+won't need to do this that often. Still - depending on how much you want to set up and debug locally, you could 
+potentially be running anywhere from 2-7 processes at once to support Lookit development.
+
+##### <a name="debugging"></a> OK this all sounds great - where do I point my debugger?
+
+If you want to debug one of the processes, you can always use PDB or the new `breakpoint` builtin. Alternatively, if
+you're using an IDE like PyCharm, you can set up handy debug profiles with neat conveniences like isolated environment
+variables and push-button functionality.
+
+If you just want to hack on the API or UIs, you only need to boot up Postgres and the web server (process #1):
+```bash
+brew services start postgresql  # Or `service postgresql start` if you're on linux
+./manage.py runserver           # Alternatively, `python manage.py runserver`
+```
+
+If you you want to run with local HTTPS (allowing webhook processing of videos), you'll need `runserver_plus` installed
+(should be in your virtual environment already if you ran `pipenv install --dev`) and a directory with cert files:
+```bash
+./manage.py runserver_plus --nopin --cert-file ${PATH_TO_YOUR_CERTFILE}
+```
+
+If you want to hack on experiment building, you'll need to boot up RabbitMQ and Docker first, and finally the celery 
+worker itself:
+```bash
+open /Applications/Docker.app                                        # If you're on Linux, just `docker`
+rabbitmq-server                                                      # You may need `sudo` for this
+celery worker --app=project --loglevel=INFO -Q builds,email,cleanup
+```
```

Comment:
Is this still about "where do I point my debugger"? It feels more like information that should be included in the setup instructions, esp. as this overlaps with things the `invoke setup` & `invoke server` tasks do.

Reply:
There is some overlap, however, that doesn't mean that both sections serve the same function. The invoke tasks were originally intended as a "button-push" startup for getting lookit up and running the first time; that's why they handle all the platform-dependent installation logic.

Debugging is a different situation where everything is already installed, and you just need to have a few specific services online and configured properly in order to poke at the system. If you're running the django test server or celery process through `pdb` or PyCharm's debugger, you don't want the whole chain of commands wrapped up in a single invocation.

To be fair - it is confusing, because Roberto went sort of halfway with factoring the tasks (some do installation _and_ serving while others do one or the other). That said, some of the invoke tasks are factored out/designed to allow for just execution of a single service, so I've put those invocations here. There's a larger conversation to be had about what exactly we want the invoke tasks to be responsible for, but that's for another ticket.